Yoast/wordpress-seo
# Add a help tab only if the post type is viewable and the metabox is enabled

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where the Yoast Columns Help appears in post lists that have no Yoast Columns. Props [glebkema](https://github.com/glebkema).

## Relevant technical choices:

The help tab is added by the [WPSEO_Yoast_Columns](https://github.com/Yoast/wordpress-seo/blob/trunk/admin/class-yoast-columns.php) class. And it's called by the class [WPSEO_Admin](https://github.com/Yoast/wordpress-seo/blob/trunk/admin/class-admin.php#L103). According to the [`plugins_loaded`](https://github.com/Yoast/wordpress-seo/blob/trunk/wp-seo-main.php#L509) hook it collects the necessary classes and calls the `register_hooks()` method for each of them.

Unfortunately, we cannot add here a check for the [`publicly_queryable`](https://developer.wordpress.org/reference/functions/register_post_type/) property of a post type, since at this moment post types are not registered yet.

Therefore, I placed all the checks to [WPSEO_Yoast_Columns](https://github.com/Yoast/wordpress-seo/blob/trunk/admin/class-yoast-columns.php), where they will work on the `load-edit.php` hook when all the information is available. For this, I added two private methods based on code from similar methods from the [WPSEO_Meta_Columns](https://github.com/Yoast/wordpress-seo/blob/trunk/admin/class-meta-columns.ph) class.

## Test instructions

This PR can be tested by following these steps:

1. Register or use 3 custom post types:
  1.1. For the first of them set the `publicly_queryable` argument as `false`.
  1.2. For the second one toggle the `Yoast SEO Meta Box` switcher to `Hide`  (`Admin Panel \ SEO \ Search Appearance \ Content Types`).
  1.3. For the third one leave both "publicly_queryable" as "true" and a switch as "View".
2. Check the `Yoast Columns` item in the `Help` tab on the post listings for each of these post types. This item (or the entire tab) should be added only for the third post type.

## UI changes

## Documentation

## Quality assurance


Fixes a part of #9215
